dvc/remote/oss.py

```diff
@@ -107,15 +107,15 @@ def list_cache_paths(self):
     def _upload(
         self, from_file, to_info, name=None, no_progress_bar=False, **_kwargs
     ):
-        cb = None if no_progress_bar else Callback(name)
-        self.oss_service.put_object_from_file(
-            to_info.path, from_file, progress_callback=cb
-        )
+        with Tqdm(desc_truncate=name, disable=no_progress_bar) as pbar:
+            self.oss_service.put_object_from_file(
+                to_info.path, from_file, progress_callback=pbar.update_to
```

Comment:
Does `disable=False` set `pbar.update_to` to `None`? And if that doesn't then does it add any delay to the operation?

Reply:
It doesn't set it to `None`. Instead there will be a negligible overhead of `tqdm` checking its `disable` state (won't print anything). Don't think there's any better approach - certainly much quicker than the previous approach!

Reply:
The progress bar overhead is about 11% [here](https://github.com/iterative/dvc/pull/2375#issuecomment-519561179), but that was our old implementation and it was not noop there. If you are sure the delay is negligible than this is a no issue.

Reply:
https://github.com/tqdm/tqdm/blob/v4.33.0/tqdm/_tqdm.py#L1108-L1110

pretty sure, yes :)